optimistic_lock.hpp

```diff
@@ -55,6 +55,9 @@ inline void spin_wait_loop_body() noexcept {
 }
 // LCOV_EXCL_STOP
 
+// The bare data for the version information on a node.
+using version_tag_type = std::uint64_t;
```

Comment:
TODO can we use `optimistic_lock::version_type` instead?

Reply:
Added TODO with your name.